Sources/Algorithms/Partition.swift

```diff
@@ -168,18 +168,18 @@ extension MutableCollection where Self: BidirectionalCollection {
 //===----------------------------------------------------------------------===//
 
 extension Collection {
-  /// Returns the index of the first element in the collection that matches
-  /// the predicate.
+  /// Returns the start index of the second partition of a collection already
+  /// partitioned according to the predicate.
```

Comment:
I agree with @xwu that we don't want to lose the more tersely stated precondition below. What would you think about the following edit to remove the redundancy?

```suggestion
  /// Returns the start index of the partition of a collection that matches 
  /// the given predicate.
```